From a6d111045e3f6f424e5a04a47d061a95e7b113e3 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:51 -0400
Subject: [tools] perf/tests: Handle throttle events in 'object code reading' test

Message-id: <1408462094-14194-303-git-send-email-jolsa@redhat.com>
Patchwork-id: 88240
O-Subject: [PATCH RHEL7.1 BZ1131394 302/325] perf tests: Handle throttle events in 'object code reading' test
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 48095b721ca54ffa70427185c00473530f4aef06
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Tue Nov 12 09:59:33 2013 +0200

description
===========
Unhandled events cause an error that fails the test, fix it.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/tests/code-reading.c b/tools/perf/tests/code-reading.c
index 6d9dc19..85d4919 100644
--- a/tools/perf/tests/code-reading.c
+++ b/tools/perf/tests/code-reading.c
@@ -275,8 +275,19 @@ static int process_event(struct machine *machine, struct perf_evlist *evlist,
  if (event->header.type == PERF_RECORD_SAMPLE)
   return process_sample_event(machine, evlist, event, state);
 
- if (event->header.type < PERF_RECORD_MAX)
-  return machine__process_event(machine, event, NULL);
+ if (event->header.type == PERF_RECORD_THROTTLE ||
+     event->header.type == PERF_RECORD_UNTHROTTLE)
+  return 0;
+
+ if (event->header.type < PERF_RECORD_MAX) {
+  int ret;
+
+  ret = machine__process_event(machine, event, NULL);
+  if (ret < 0)
+   pr_debug("machine__process_event failed, event type %u\n",
+     event->header.type);
+  return ret;
+ }
 
  return 0;
 }
-- 
1.7.1